aggsender/aggsender.go

```diff
@@ -94,11 +127,13 @@ func (a *AggSender) Start(ctx context.Context) {
 func (a *AggSender) checkInitialStatus(ctx context.Context) {
 	ticker := time.NewTicker(a.cfg.DelayBeetweenRetries.Duration)
 	defer ticker.Stop()
-
+	a.status.Status = types.StatusCheckingInitialStage
 	for {
 		if err := a.checkLastCertificateFromAgglayer(ctx); err != nil {
```

Comment:
If I can suggest to change this code a bit:

```suggestion
		err := a.checkLastCertificateFromAgglayer(ctx)
		a.status.SetLastError(err)
		if err != nil {
		    a.log.Errorf("error checking initial status: %w, retrying in %s", err, a.cfg.DelayBeetweenRetries.String())
		} else {
		    a.log.Info("Initial status checked successfully")
		    return
		}
```